File: Source/TurnTracker.Client/src/app/auth/models/Device.ts
import { DateTime } from 'luxon';
import { Session } from './Session';

export class Device {
    public name: string;
    public created: string;
    public updated: string;
    public id: number;
    public current: boolean;
    public sessions: Session[];

  // Properties populated by the client and not the server
  public createdDate: DateTime;
  public updatedDate: DateTime;
}

File: Source/TurnTracker.Domain/Models/Device.cs
using System;
using System.Collections.Generic;

namespace TurnTracker.Domain.Models
{
    public class Device
    {
        public string Name { get; set; }
        public DateTimeOffset Created { get; set; }
        public DateTimeOffset Updated { get; set; }
        public int Id { get; set; }
        public bool Current { get; set; }
        public IEnumerable<Session> Sessions { get; set; }
    }
}